### PR TITLE
Swipable switch

### PR DESCRIPTION
**Pull request by Nevon**
swipeable-switch

The code isn't the most beautiful thing I've ever written, but it works quite well. Feels A LOT better now than it did before.

The current implementation allows you to start swiping anywhere on the label or the bullet, because the bullet by itself is a rather small target.

Something to consider:

* We had an off-by-one error where you could swipe it 1px "too far", and then it would snap back to the right edge when you let go. I actually felt like this felt really good and tactile, but I'm not sure if that's the direction we want to go with these components.

### CODE REVIEW

File: Switch/Toggle/index.jsx [hunk 1]
```diff
@@ -86,11 +166,11 @@ const Toggle = React.createClass({
       ...remainingProps
     } = this.props
 
-    const { pressed } = this.state
+    const { pressed, pseudoValue } = this.state
 
     const classNames = classNamesBind.bind({ ...defaultStyles, ...styles })
     const cls = classNames(baseClass, {
-      'is-checked': value,
+      'is-checked': pseudoValue != null ? pseudoValue : value,
```
Comment (joaomilho) [inline Switch/Toggle/index.jsx]: `!==` ?

File: Switch/Toggle/styles.scss [hunk 2]
```diff
@@ -86,40 +97,11 @@
...
-
-  .right.is-checked & {
-    left: auto;
-    right: ($grid * .2);
-  }
-
-  .right.is-pressed.is-checked & {
-    left: auto;
-    right: ($grid * .2);
+    transform: translateX($grid * 3);
```
Comment (joaomilho) [inline Switch/Toggle/styles.scss]: IE9?

**Comment by joaomilho** [comment]
awesome

File: Switch/Toggle/index.jsx (hunk 1 above)
Comment (Nevon) [inline Switch/Toggle/index.jsx]: The value would be `undefined`, which wouldn't match. `!=` matches both `null` and `undefined`. However, there shouldn't be a scenario where the value is `null`, so we might as well do `!== undefined`.

**Comment by xaviervia** [comment]
@Alexintrino pointed out that we might want to block the scroll event when doing that, since otherwise it could back in the history and so on.

**Comment by Nevon** [comment]
I don't think we actually need to. I've tested it out in both iOS and Android, and the only way it would happen is if you position the switch completely flush against the edge of the screen. In the showroom where it's 20 or so pixels off from the side, I never accidentally tripped the history back function.

**Comment by xaviervia** [comment]
Awesome

**Comment by xaviervia** [comment]
I want to do a quick visual review at this with the designers today since among other things we killed the making-it-wider-on-press. Other than that I'm super happy about it, I think we can iterate on the specific handler code later on when we have the need for other components capturing swipe.

**Comment by Nevon** [comment]
Live demo for the designers to review is available here: https://jsfiddle.net/rso7uh0v/embedded/result/dark/

Open on an actual phone or use touch emulation in the browser for the touch interactions to work.

**Comment by xaviervia** [comment]
Awesome. @hannah-ahl care to review? The change to notice (aside from the sliding) is that the dot in the switch no longers get fat when pressed. Is that OK with you?

**Comment by Nevon** [comment]
So Corbett immediately managed to do the history back gesture as a native iOS user. 😢 The back gesture is implemented in the OS level, so there's no way for us to cancel it (thanks Apple).

The only way we can work around it is to design our interfaces in such a way that the switch is not close to the edge. For situations where the control will be close to the edge, you should use a checkbox.

**Comment by xaviervia** [comment]
Oj, oj, oj, oj

**Comment by hannah-ahl** [comment]
I’d probably ask Jordi or Alexey for this specific case, I find myself too unfamiliar with how the switch worked before to be able to give a judgement
Hannah Ahl
Product Designer
Product
+46 700 029 449 <tel:+46+700+029+449>
Klarna AB
Sveavägen 46, 111 34 Stockholm
Tel: +46 8 120 120 00 <tel:+46812012000>
Reg no: 556737-0431
klarna.co <http://klarna.com/>m

**Comment by hannah-ahl** [comment]
Gosh darn it
Hannah Ahl
Product Designer
Product
+46 700 029 449 <tel:+46+700+029+449>
Klarna AB
Sveavägen 46, 111 34 Stockholm
Tel: +46 8 120 120 00 <tel:+46812012000>
Reg no: 556737-0431
klarna.co <http://klarna.com/>m

**Pull request merged**